### PR TITLE
docs(api): Versioning page for Python API v2.15

### PR DESCRIPTION
**Pull request by ecormany**
# Overview

Updates to the Versioning page for Python Protocol API v2.15.

Restructured the page so newer versions are at the top.

# Test Plan

Check [the sandbox](http://sandbox.docs.opentrons.com/docs-versioning-2.15/v2/versioning.html).

# Changelog

- adds 2.15 release notes
- flips order of version table and changelog section
- explains putting `apiLevel` in  `metadata` vs. `requirements` 
  - adds anchor in `tutorial.rst` for cross-reference
- gives instructions for newer app versions first
- lists supported versions for Flex and OT-2
- standardizes RST header styles

# Review requests

Are there any other additions, removals, or fixes for 2.15 that need to be included?

Is the supported version range for Flex correct? cf. RSS-264.

# Risk assessment

none, docs only.

### CODE REVIEW

File: api/docs/v2/versioning.rst [hunk 1]
```diff
@@ -2,13 +2,14 @@
...
+**********
 
-The Python Protocol API has its own versioning system, which is separate from the versioning system used for the robot software and the Opentrons App. This allows you to specify the API version that your protocol requires without worrying about what robot software versions it will work with.
+The Python Protocol API has its own versioning system, which is separate from the versioning system used for the robot software and the Opentrons App. This allows protocols to run on newer robot software versions without modification.
 
 Major and Minor Versions
-------------------------
+========================
 
 The API uses a major and minor version number and does not use patch version numbers. For instance, major version 2 and minor version 0 is written as ``2.0``. Versions are not decimal numbers, so ``2.10`` indicates major version 2 and minor version 10, which is 9 minor versions newer than ``2.1``. The Python Protocol API version will only increase based on changes that affect protocol behavior.
```
Comment (jwwojak) [inline api/docs/v2/versioning.rst]: Consider dropping the last part of this sentence, I think it adds more than needed:

>  Versions are not decimal numbers, so ``2.10`` indicates major version 2 and minor version 10 ~~, which is 9 minor versions newer than ``2.1``~~

That seems sufficiently clear to me.

File: api/docs/v2/versioning.rst [hunk 2]
```diff
@@ -32,294 +33,345 @@ You must specify the API version you are targeting in the ``metadata`` block at
...
 
-From version 3.15.0 to 5.0.2 of the OT-2 software and Opentrons App, the maximum supported API level was listed in the robot's **Information** card in the app. Since version 6.0.0, the same information is listed under **Robot Settings > Advanced**.
 
-If you upload a protocol that specifies a higher API level than the maximum supported by your OT-2, it won't be able to analyze or run your protocol. You may be able to increase the maximum supported version by updating your OT-2 to the latest robot software and Opentrons App. All OT-2 hardware is capable of running protocols specifying any minor version of the API v2, as long as its software is up to date.
+Maximum Supported Versions
+==========================
 
+The maximum supported API version for your robot is listed in the Opentrons App under **Robots** > your robot > **Robot Settings** > **Advanced**. Before version 6.0.0 of the app, the same information was listed on your robot's **Information** card.
```
Comment (jwwojak) [inline api/docs/v2/versioning.rst:68]: This one is tricky. While using **bold** is correct for indicating UI components users interact with, the "your robot" in plain text kinda threw me off. Can you try **your robot** or **robot name here**? I think the text formatting should match the rest of the bold words in the path.

File: api/docs/v2/versioning.rst [hunk 3]
```diff
@@ -32,294 +33,345 @@ You must specify the API version you are targeting in the ``metadata`` block at
...
+=======================
+
+Version 2.15
+------------
+
+This version introduces support for the Opentrons Flex robot, instruments, modules, and labware.
+
+- Flex features
+
+  - Write protocols for Opentrons Flex by declaring ``"robotType": "Flex"`` in the new ``requirements`` dictionary.
```
Comment (jwwojak) [inline api/docs/v2/versioning.rst]: Suggest a link to the Tutorial section. There's a write up/explanation of how to use `metadata` and `requirements` there too.

**Review by jwwojak — APPROVED** [review]
Recommend approving. My suggestions aren't dealbreakers, though I do strongly encourage a link back to the Tutorial page regarding an explanation about `requirements` and `metadata` use.

File: api/docs/v2/versioning.rst (hunk 2 above)
Comment (ecormany) [reply api/docs/v2/versioning.rst:68]: Hmm. I've used this format in other places. I don't want people to be looking for the string "your robot" — which they won't find anywhere in the UI.

File: api/docs/v2/versioning.rst (hunk 3 above)
Comment (ecormany) [reply api/docs/v2/versioning.rst]: Added an anchor and linked. ⚓

**Pull request merged**